akka-http-server/src/main/scala/endpoints/akkahttp/routing/BasicAuthentication.scala

```diff
@@ -0,0 +1,36 @@
+package endpoints.akkahttp.routing
+
+import java.util.Base64
+
+import akka.http.scaladsl.model.headers.{Authorization, BasicHttpCredentials}
+import akka.http.scaladsl.model.{HttpHeader, HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.Directives
+import endpoints.algebra
+import endpoints.algebra.BasicAuthentication.Credentials
+
+trait BasicAuthentication extends algebra.BasicAuthentication with Endpoints {
+
+  /**
+    * Extracts the credentials from the request headers.
+    * In case of absence of credentials rejects request
+    */
+  private[endpoints] lazy val basicAuthentication: RequestHeaders[Credentials] =
+    Directives.headerValue(extractCredentials)
```

Comment:
Sure, with basic auth it makes sense. I will fix it and document in `algebra.BasicAuthentication` and we can discuss general rules for headers and entities outside of this PR, ok? I know it may introduce inconsistency between interpreters, but it's probably not so big problem right now and without automatic testing it will be very difficult to make sure that they are fully consistent.